### PR TITLE
Fix filtering by greater/lesser/etc

### PR DESCRIPTION
**Pull request by xronos-i-am**

### CODE REVIEW

File: app/assets/javascripts/activeadmin_addons/select2.js [hunk 1]
```diff
@@ -62,6 +62,17 @@ $(function() {
...
         });
+
+        var $parent = $(el).parent(), $label = $parent.find('label');
+
+        if ($label.length) {
+          var $input = $parent.find('#' + $label.attr('for'))
+
+          $(el).on('change', function(value) {
+            $input.attr('name', 'q[' + $(this).val() + ']')
+          })
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/select2.js]: Missing semicolon                                                  semi

File: app/assets/javascripts/activeadmin_addons/select2.js [hunk 2]
```diff
@@ -62,6 +62,17 @@ $(function() {
...
           allowClear: allowClear
         });
+
+        var $parent = $(el).parent(), $label = $parent.find('label');
+
+        if ($label.length) {
+          var $input = $parent.find('#' + $label.attr('for'))
+
+          $(el).on('change', function(value) {
+            $input.attr('name', 'q[' + $(this).val() + ']')
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/select2.js]: Unexpected string concatenation                                    prefer-template<br>'$' is not defined                                                 no-undef<br>Missing semicolon                                                  semi

File: app/assets/javascripts/activeadmin_addons/select2.js [hunk 3]
```diff
@@ -62,6 +62,17 @@ $(function() {
           width: 'resolve',
           allowClear: allowClear
         });
+
+        var $parent = $(el).parent(), $label = $parent.find('label');
+
+        if ($label.length) {
+          var $input = $parent.find('#' + $label.attr('for'))
+
+          $(el).on('change', function(value) {
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/select2.js]: '$' is not defined                                                 no-undef<br>Missing space before function parentheses                          space-before-function-paren<br>'value' is defined but never used                                  no-unused-vars

File: app/assets/javascripts/activeadmin_addons/select2.js [hunk 4]
```diff
@@ -62,6 +62,17 @@ $(function() {
           width: 'resolve',
           allowClear: allowClear
         });
+
+        var $parent = $(el).parent(), $label = $parent.find('label');
+
+        if ($label.length) {
+          var $input = $parent.find('#' + $label.attr('for'))
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/select2.js]: Unexpected var, use let or const instead                           no-var<br>All 'var' declarations must be at the top of the function scope    vars-on-top<br>Unexpected string concatenation                                    prefer-template<br>Missing semicolon                                                  semi

File: app/assets/javascripts/activeadmin_addons/select2.js [hunk 5]
```diff
@@ -62,6 +62,17 @@ $(function() {
           width: 'resolve',
           allowClear: allowClear
         });
+
+        var $parent = $(el).parent(), $label = $parent.find('label');
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/select2.js]: Unexpected var, use let or const instead                           no-var<br>All 'var' declarations must be at the top of the function scope    vars-on-top<br>Split 'var' declarations into multiple statements                  one-var<br>'$' is not defined                                                 no-undef<br>Expected variable declaration to be on a new line                  one-var-declaration-per-line

File: app/assets/javascripts/activeadmin_addons/select2.js [hunk 6]
```diff
@@ -62,6 +62,19 @@ $(function() {
...
+
+        if ($label.length) {
+          var $input = $parent.find('#' + $label.attr('for'))
+
+          if (!$input.attr('id') == $(el).attr('id')) {
+            $(el).on('change', function(value) {
+              $input.attr('name', 'q[' + $(this).val() + ']')
+            })
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/select2.js]: Missing semicolon                                                  semi

File: app/assets/javascripts/activeadmin_addons/select2.js [hunk 7]
```diff
@@ -62,6 +62,19 @@ $(function() {
...
         });
+
+        var $parent = $(el).parent(), $label = $parent.find('label');
+
+        if ($label.length) {
+          var $input = $parent.find('#' + $label.attr('for'))
+
+          if (!$input.attr('id') == $(el).attr('id')) {
+            $(el).on('change', function(value) {
+              $input.attr('name', 'q[' + $(this).val() + ']')
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/select2.js]: Unexpected string concatenation                                    prefer-template<br>'$' is not defined                                                 no-undef<br>Missing semicolon                                                  semi

File: app/assets/javascripts/activeadmin_addons/select2.js [hunk 8]
```diff
@@ -62,6 +62,19 @@ $(function() {
...
           allowClear: allowClear
         });
+
+        var $parent = $(el).parent(), $label = $parent.find('label');
+
+        if ($label.length) {
+          var $input = $parent.find('#' + $label.attr('for'))
+
+          if (!$input.attr('id') == $(el).attr('id')) {
+            $(el).on('change', function(value) {
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/select2.js]: '$' is not defined                                                 no-undef<br>Missing space before function parentheses                          space-before-function-paren<br>'value' is defined but never used                                  no-unused-vars

File: app/assets/javascripts/activeadmin_addons/select2.js [hunk 9]
```diff
@@ -62,6 +62,19 @@ $(function() {
           width: 'resolve',
           allowClear: allowClear
         });
+
+        var $parent = $(el).parent(), $label = $parent.find('label');
+
+        if ($label.length) {
+          var $input = $parent.find('#' + $label.attr('for'))
+
+          if (!$input.attr('id') == $(el).attr('id')) {
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/select2.js]: Expected '===' and instead saw '=='                                eqeqeq<br>'$' is not defined                                                 no-undef

**Comment by ldlsegovia** [comment]
@xronos-i-am I don't understand what are you fixing here. Can you explain it to me?

**Comment by xronos-i-am** [comment]
@ldlsegovia there are condition modifiers in the "Filters" menu (lesser, greater, equal, contains, etc). The commit fixes it. Condition should change on change value the corresponding select tag

**Comment by ldlsegovia** [comment]
@xronos-i-am I don't know how this is failing...

search

**Comment by xronos-i-am** [comment]
@ldlsegovia i have to use activeadmin-1.0.0.pre, probably this is the reason you can't see the issue

**Pull request closed without merging**